codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -215,9 +216,7 @@ public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf dat
             // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
 
-            verifyEndOfStreamNotReceived(stream);
             verifyGoAwayNotReceived();
-            verifyRstStreamNotReceived(stream);
 
             // We should ignore this frame if RST_STREAM was sent or if GO_AWAY was sent with a
             // lower stream ID.
```

Comment:
I wanted to leave a comment on [line 233](https://github.com/netty/netty/pull/3386/files#diff-179f52a9de82166ed577701a1412b5a0R233)...but too far from modifications.  If the stream state is HALF_CLOSED_REMOTE does it matter if we sent reset or not?  The remote side shouldn't be sending data if it has sent END_STREAM flag right?

[section 5.1](https://tools.ietf.org/html/draft-ietf-httpbis-http2-16#section-5.1):
>half closed (remote):
...
> If an endpoint receives additional frames for a stream that is in
      this state, other than WINDOW_UPDATE, PRIORITY or RST_STREAM, it
      MUST respond with a stream error (Section 5.4.2) of type
      STREAM_CLOSED.